### PR TITLE
deps: Fix nightly dependency bot job

### PR DESCRIPTION
**Pull request by perlpunk**
Before, the job was running a docker in the already running docker.
CircleCI doesn't allow mounting volumes in that scenario, that's why the
`make update-deps` and `tools/tidy` never worked.
We just didn't see that because we never had a perltidy update since that
code was added.

This commit simply runs the actual job in the already running docker.

It only needs to install `hub` to make the actual PR (in case of changes).

So we don't need an extra image for the bot, but I left the file in place so
far:

    docker/devel:openQA:ci/dependency_bot/Dockerfile

I tested it in a branch in the origin repo to see if it created a PR: https://github.com/os-autoinst/openQA/pull/3248

I also converted line endings in `.circleci/ci-packages.txt` with `dos2unix`.

### CODE REVIEW

File: .circleci/build_dependencies.sh [hunk 1]
```diff
@@ -15,39 +15,34 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-set -e
+set -ex
```
Comment (okurz) [inline .circleci/build_dependencies.sh:18]: I recommend to not use tracing by default in this script. The `-x` can still be added where needed, e.g. within the calling location

**Comment by perlpunk** [comment]
> I would prefer to keep a separate container image with the tools so that no additional packages need to be reinstalled on every call.

The whole purpose of this PR is to only use only one image, because in CircleCI one cannot use mounted volumes in a nested docker call.
Where is the problem with installing `hub`? And it's not done on every call, only when there are changes.
We are installing around 250 packages in this CI job every night.
Why is installing `hub` problematic?

**Comment by Martchus** [comment]
I also don't see how it is problematic.

File: .circleci/build_dependencies.sh (hunk 1 above)
Comment (perlpunk) [reply .circleci/build_dependencies.sh]: removed it

**Comment by perlpunk** [comment]
I split the `dos2unix` into its own commit.

**Pull request merged**